mrmustard/lab_dev/states/dm.py

```diff
@@ -130,22 +130,26 @@ def from_fock(
     @classmethod
     def from_ansatz(
         cls,
-        modes: Sequence[int],
+        modes: Iterable[int],
         ansatz: PolyExpAnsatz | ArrayAnsatz | None = None,
         name: str | None = None,
     ) -> State:
+        if not isinstance(modes, set) and sorted(modes) != list(modes):
+            raise ValueError(f"Modes must be sorted. got {modes}")
```

Comment:
Seeing as we are already sorting the modes to do the check does it make sense to just use that?

Reply:
it was to avoid users inputting modes with a particular order and expecting DM to retain it

Reply:
We could either or alternatively we could throw a warning. We could even just document "modes will be sorted in order".